lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/Predict/SomaticValidation.pm

```diff
@@ -0,0 +1,247 @@
+
+sub execute {
+    my $self = shift;
+
+    my $analysis_project = $self->analysis_project;
+
+    my $normal_sample_common_names = $self->normal_sample_common_names;
+    my $tumor_sample_common_names  = $self->tumor_sample_common_names;
+
+    my $id_method = $self->sample_identifier;
+
+    my $samples_by_individual_id = $self->resolve_samples_by_individual_id;
+    my @individual_ids = sort keys %{$samples_by_individual_id};
+    $self->status_message('Found '. scalar(@individual_ids) .' individuals');
+
+    for my $individual_id (@individual_ids) {
+        my $individual = Genome::Individual->get($individual_id);
+
+        my @samples = @{$samples_by_individual_id->{$individual_id}};
+
+        my $sample_set = Set::Scalar->new(@samples);
+        $self->status_message('Found '. $sample_set->size .' samples for individual '. $individual->__display_name__);
+
+        my @normal_samples;
+        my @tumor_samples;
+
+        for my $sample ($sample_set->members) {
+            my $sample_common_name = $sample->common_name || 'NULL';
+            push(@normal_samples, $sample) if ($sample_common_name =~ /$normal_sample_common_names/i);
+            push(@tumor_samples,  $sample) if ($sample_common_name =~ /$tumor_sample_common_names/i);
+        }
+
+        my $normal_sample_set = Set::Scalar->new(@normal_samples);
+        my $tumor_sample_set = Set::Scalar->new(@tumor_samples);
+
+        if ($normal_sample_set->size > 1) {
+            $self->fatal_message('The following samples for individual '. $individual->__display_name__ .' are ALL found to be normal: '. join(',', map {$_->$id_method} $normal_sample_set->members));
+        }
+        
+        my $unique_sample_set = $sample_set->unique($normal_sample_set,$tumor_sample_set);
+        if ($unique_sample_set) {
+            $self->fatal_message('The following samples for individual: '. $individual->__display_name__ .' do not match either tumor or normal criteria: '. join(',', map{$_->$id_method} $unique_sample_set->members));
+        }
+
+        my $intersection_sample_set = $normal_sample_set->intersection($tumor_sample_set);
+        if ($intersection_sample_set) {
+            $self->fatal_message('The following samples for individual '. $individual->__display_name__ .' meet both tumor and normal match criteria: '. join(',', map {$_->$id_method} $intersection_sample_set->members));
+        }
+
+        if ($normal_sample_set && $tumor_sample_set) {
+            my ($normal_sample) = $normal_sample_set->members();
+            for my $tumor_sample ($tumor_sample_set->members) {
+                my %data = (
+                    tumor_sample => $tumor_sample->$id_method,
+                    normal_sample => $normal_sample->$id_method,
+                    snv_variant_list_id => '',
+                    indel_variant_list_id => '',
+                    sv_variant_list_id => '',
+                    tag => 'somatic',
```

Comment:
I went a little further and added this to add_subject_mapping.  It will create the data hash and add it to the new set.